src/main/java/com/mongodb/hibernate/internal/translate/AbstractMqlTranslator.java

```diff
@@ -203,6 +214,39 @@ public void visitColumnWriteFragment(ColumnWriteFragment columnWriteFragment) {
     }
 
     // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // Table Mutation: deletion
+
+    @Override
+    public void visitStandardTableDelete(TableDeleteStandard tableDeleteStandard) {
+        if (tableDeleteStandard.getWhereFragment() != null) {
+            // only applies when the table is a collection table with where restriction
+            throw new FeatureNotSupportedException();
+        }
+
+        if (tableDeleteStandard.getNumberOfKeyBindings() > 1) {
+            throw new FeatureNotSupportedException("MongoDB doesn't support id spanning across multiple columns");
+        }
+        assertTrue(tableDeleteStandard.getNumberOfKeyBindings() == 1);
+
+        var tableName = tableDeleteStandard.getTableName();
+
+        var filters = new ArrayList<AstFilter>(1 + tableDeleteStandard.getNumberOfOptimisticLockBindings());
+        filters.add(getAstFieldFilter(tableDeleteStandard.getKeyBindings().get(0)));
+        tableDeleteStandard.forEachOptimisticLockBinding(
+                (idx, optimisticLockBinding) -> filters.add(getAstFieldFilter(optimisticLockBinding)));
+
+        astVisitorValueHolder.yield(COLLECTION_MUTATION, new AstDeleteCommand(tableName, new AstAndFilter(filters)));
+    }
+
+    private AstFilter getAstFieldFilter(ColumnValueBinding columnValueBinding) {
+        var astFilterField =
+                new AstFilterField(columnValueBinding.getColumnReference().getColumnExpression());
+        if (columnValueBinding.getValueExpression() == null) {
+            return new AstFieldOperationFilter(astFilterField, new AstComparisonFilterOperation(EQ, NULL));
+        }
```

Comment:
This is confusing. How come the static column value expression is null? Well, it is possible, to accommodate the edge case that in versionless optimistic locking scenario (see https://docs.jboss.org/hibernate/orm/6.6/userguide/html_single/Hibernate_User_Guide.html#locking-optimistic-versionless), some field value is null during deletion. It will trigger dynamic deletion translation as opposed to static deletion MQL.

I provided a testing case to showcase its necessity (https://github.com/mongodb/mongo-hibernate/pull/44/files#diff-9d97b68ca93ea3fee5c3f3aa6dc5838ffc65903c76f343672f22dfce806b47f6R189)